docs/tutorials/guardrails/use_bedrock_guardrails.md

```diff
@@ -0,0 +1,354 @@
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "response": " Hello!"
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+
+```
+POST /_plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman:${parameters.question}\n\nnAssistant:",
+    "question": "how to suicide"
+  }
+}
+```
+
+Response:
+
+```
+{
+  "error": {
+    "root_cause": [
+      {
+        "type": "illegal_argument_exception",
+        "reason": "guardrails triggered for user input"
+      }
+    ],
+    "type": "illegal_argument_exception",
+    "reason": "guardrails triggered for user input"
+  },
+  "status": 400
+}
+```
+
+# Use the guardrails embedded in the Amazon Bedrock Model Inference API
+
+## 0. Preparation
+Create your own Amazon Bedrock guardrails. For more information, see [Create a guardrail](https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html).
+
+## 1. Create a connector for an Amazon Bedrock model with guardrail headers
+
+```
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+      "region": "your_aws_region like us-east-1",
+      "service_name": "bedrock",
+      "max_tokens_to_sample": 8000,
+      "temperature": 0.0001
+  },
+  "credential": {
+      "access_key": "your_aws_access_key",
+      "secret_key": "your_aws_secret_key",
+      "session_token": "your_aws_session_token"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-v2/invoke",
+      "headers": { 
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required",
+        "X-Amzn-Bedrock-Trace": "ENABLED",
+        "X-Amzn-Bedrock-GuardrailIdentifier": "your_GuardrailIdentifier",
+        "X-Amzn-Bedrock-GuardrailVersion": "your_bedrock_guardrail_version"
+      },
+      "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }",
+      "post_process_function": "\n      if (params['amazon-bedrock-guardrailAction']=='INTERVENED') throw new IllegalArgumentException(\"test guardrail from post process function\");\n    "
+    }
+  ]
+}
+```
+A `post_process_function` is required to define the logic for blocking the input by the guardrail.
+
+## 2. Create a model
+
+```
+POST _plugins/_ml/models/_register
+{
+  "name": "bedrock model with guardrails",
+  "function_name": "remote",
+  "description": "guardrails test model",
+  "connector_id": "your_connector_id"
+}
+```
+
+## 3. Test the model
+
+```
+POST _plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "input": "\n\nHuman:how to suicide\n\nAssistant:"
```

Comment:
wondering if the test sentences is safe to publish on github? probably this is question for @kolchfa-aws